src/helper/engine.cpp

```diff
@@ -88,6 +89,7 @@ std::optional<engine::result> engine::context::publish(parameter &&param)
                     std::make_move_iterator(event->data.begin()),
                     std::make_move_iterator(event->data.end()));
                 event_actions.merge(event->actions);
+                schemas.merge(event->schemas);
```

Comment:
Maybe we should simplify this even further since the WAF is the only consumer... (just making a comment, not asking you to change it)